sky/core.py

```diff
@@ -552,14 +555,12 @@ def cancel(
         sky.exceptions.CloudUserIdentityError: if we fail to get the current
           user identity.
     """
-    if not job_ids and not all:
-        raise ValueError(
-            'sky cancel requires either a job id '
-            f'(see `sky queue {cluster_name} -s`) or the --all flag.')
-
     backend_utils.check_cluster_name_not_reserved(
         cluster_name, operation_str='Cancelling jobs')
 
+    if cancel_latest_running_job:
+        job_ids = None
```

Comment:
We probably should raise the `ValueError` when `job_ids` is not None and `cancel_latest_running_job` (or `cancel_all` is set), to avoid ambiguous behavior.